peps/pep-0747.rst

```diff
@@ -569,19 +535,35 @@ Subtyping
 Whether a ``TypeExpr`` value can be assigned from one variable to another is
 determined by the following rules:
 
+Relationship with type
+''''''''''''''''''''''
+
 ``TypeExpr[]`` is covariant in its argument type, just like ``type[]``:
 
 -  ``TypeExpr[T1]`` is a subtype of ``TypeExpr[T2]`` iff ``T1`` is a
    subtype of ``T2``.
 -  ``type[C1]`` is a subtype of ``TypeExpr[C2]`` iff ``C1`` is a subtype
    of ``C2``.
 
-A plain ``type`` can be assigned to a plain ``TypeExpr`` but not the
-other way around:
+An unparameterized ``type`` can be assigned to an unparameterized ``TypeExpr``
+but not the other way around:
 
 -  ``type[Any]`` is assignable to ``TypeExpr[Any]``. (But not the
    other way around.)
 
+Relationship with UnionType
+'''''''''''''''''''''''''''
+
+``TypeExpr[U]`` is a subtype of ``UnionType`` iff ``U`` is
```

Comment:
> There are many ways that union types can be formed in static analysis. For example, they arise from "joins" in code flow. This doesn't necessarily mean that at runtime the value is implemented with UnionType.

Interesting. It sounds like I'll have to remove the following rule entirely:

* ✖️ TypeExpr[U] is a subtype of UnionType iff U is the type expression X | Y | ...

A consequence of that is that a function parameter annotated as `UnionType` will no longer be able to accept type expressions in the form `X | Y` because that is now treated as a `TypeExpr[X | Y]` rather than a `UnionType` and there is no way assign a `TypeExpr[X | Y]` to a `UnionType`. **So this would be one scenario where backward compatiblity would be intentionally broken.** Hopefully this scenario is uncommon (outside of typeshed):

```
def i_like_unions(t: UnionType) -> None:
    pass

i_like_unions(int | str)  # accepted before PEP 747; rejected after PEP 747
```